File: HorseRacing.Contracts/Models/User/Requests/GetUser/GetUserRequest.cs
using HorseRacing.Contracts.Base.Requests;

namespace HorseRacing.Contracts.Models.User.Requests.GetUser
{
    public class GetUserRequest : BaseRequest<GetUserRequestDto>
    {
        public GetUserRequest() : base() { }
        public GetUserRequest(GetUserRequestDto data) : base(data) { }
    }
}

File: HorseRacing.Web/src/interfaces/api/contracts/model/user/requests/get-user/get-user-request.ts
import type { BaseRequest } from '~/interfaces/api/contracts/base/requests/base-request';
import type { GetUserRequestDto } from './get-user-request-dto';

export class GetUserRequest implements BaseRequest<GetUserRequestDto> {
    Data: GetUserRequestDto;

    constructor(data?: GetUserRequestDto) {
        this.Data = data || { UserId: '' };
    }
}
